File: programmers/공통/[5회차]42746. 가장 큰 수/한경/가장 큰 수.java
import java.util.Arrays;

class Solution {
   static String solution(int[] numbers) {
		
		String[] str = new String[numbers.length];
 
        for(int i = 0; i < numbers.length; i++){
            // string 으로 변경하여 배열에 삽입
            str[i] = String.valueOf(numbers[i]);
        }
 
        // 두 수를 합친 값이 큰 순서대로 내림차순 정렬
        Arrays.sort(str, (o1, o2) -> (o2 + o1).compareTo(o1 + o2));
 
        // 첫 번째 수가 0이면 0을 리턴.
        if (str[0].equals("0")) {
            return "0";
        }
 
        // 문자열 연결하여 return
        return String.join("", str);
	}
}

File: programmers/공통/[5회차] 42746. 가장 큰 수/창하/가장 큰 수.js
function solution(numbers) {

    var result = numbers
       .sort(function(a,b){
       if( ('' + b + b + b).substring(0,3) == ('' + a + a + a).substring(0,3)) return ('' + b + a) - ('' + a + b);
       return ('' + b + b + b).substring(0,3) - ('' + a + a + a).substring(0,3) })
       .join('');
    
    return Math.max(...numbers) == 0 ? '0' : result;
}



function solution(numbers) {

    var result = numbers
        .sort(function(a,b) {
            return `${b}${a}` - `${a}${b}`;
        })
        .join('');

    return result[0] === '0' ? '0' : result;
}